frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -40,7 +41,17 @@ const CloudShellTerminal: React.FC<CloudShellTerminalProps &
     verb: 'create',
     resource: 'pods',
   });
-  const [workspace, loaded, loadError] = useCloudShellWorkspace(user, isAdmin, namespace);
+  const [isv1Alpha2Available, isv1alpha2CheckLoading] = CheckV1alpha2CRDAvailability();
+  let workspaceModel = WorkspaceModel;
+  if (!isv1Alpha2Available) {
+    workspaceModel = v1alpha1WorkspaceModel;
+  }
```

Comment:
```suggestion
  const workspaceModel = React.useMemo(() => (!isv1Alpha2Available ? v1alpha1WorkspaceModel : WorkspaceModel), [isv1Alpha2Available])
```

Reply:
A memo here is overkill as there's no computed value here that requires a stable reference. It's simply an if condition to reassign the `workspaceModel` value.